### DISCUSSION TITLE
e2e testing approach

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#133] [General]
For e2e tests, the Helm Operator and Flux use:

1. `bats` (testing framework)
2. `make e2e` (allows running locally, as opposed to github actions only)
3. new kind cluster for each test (test isolation and potentially parallelization)

Are these all still target state for the toolkit components? Or is there an alternate approach planned?

**Comment by phillebaba** [comment]
Something else that should also be looked at in this case is Flagger as it [also supports e2e testing](https://docs.flagger.app/usage/webhooks#integration-testing). It would be good to have a standardized method of running tests on deployment that is the same for Helm Controller Kustomize Controller and Flagger.

**Comment by seaneagan** [comment]
For e2e test isolation and parallelism, could just use github workflow job matrix. This doesn't help with local runs, but maybe that can come later?
